projects/js-packages/components/components/boost-score-graph/uplot-line-chart.tsx

```diff
@@ -0,0 +1,148 @@
+import { __ } from '@wordpress/i18n';
+import React, { useState, useMemo, useRef, useCallback } from 'react';
+import uPlot from 'uplot';
+import UplotReact from 'uplot-react';
+import { getUserLocale } from '../../lib/locale';
+import numberFormat from '../number-format';
+import getDateFormat from './get-date-format';
+import getPeriodDateFormat from './get-period-date-format';
+import { tooltipsPlugin } from './tooltips-plugin';
+import useResize from './use-resize';
+
+import 'uplot/dist/uPlot.min.css';
+
+const DEFAULT_DIMENSIONS = {
+	height: 300,
+	width: 600,
+};
+
+interface UplotChartProps {
+	data: uPlot.AlignedData;
+	options?: Partial< uPlot.Options >;
+	legendContainer?: React.RefObject< HTMLDivElement >;
+	solidFill?: boolean;
+	period?: string;
+}
+
+/**
+ * Creates a series information object for uPlot based on the label and color.
+ *
+ * @param {string} label - The label for the series.
+ * @param {string} color - The color of the series.
+ * @returns {object} The series information object.
+ */
+function createSerieInfo( label: string, color: string ) {
+	const { spline } = uPlot.paths;
+	return {
+		label: label,
+		stroke: color,
+		width: 2,
+		paths: ( u, seriesIdx, idx0, idx1 ) => {
+			return spline?.()( u, seriesIdx, idx0, idx1 ) || null;
+		},
+		points: {
+			show: false,
+		},
+		value: ( self: uPlot, rawValue: number ) => {
+			if ( ! rawValue ) {
+				return '-';
+			}
+
+			return numberFormat( rawValue );
+		},
+	};
+}
+
+/**
+ * UplotLineChart component.
+ *
+ * @param {object} root0 - The props object for the UplotLineChart component.
+ * @param {uPlot.AlignedData} root0.data - The data for the uPlot chart.
+ * @param {string} root0.period - The period to be used for the chart.
+ * @returns {React.Element} The JSX element representing the UplotLineChart component.
+ */
+export default function UplotLineChart( { data, period }: UplotChartProps ) {
+	const uplot = useRef< uPlot | null >( null );
+	const uplotContainer = useRef( null );
+
+	const [ options ] = useState< uPlot.Options >(
+		useMemo( () => {
```

Comment:
Thanks for pointing this out.
I removed the `useState` in favor of only using `useMemo` because on our case we won't modify the options at all:
https://github.com/Automattic/jetpack/pull/32016/commits/216b854112ea9fb3be1b05b89b3af7da6f1a996b